xbmc/addons/AddonDatabase.cpp

```diff
@@ -871,7 +878,30 @@ bool CAddonDatabase::DisableAddon(const std::string &addonID, bool disable /* =
     if (!m_pDS)
       return false;
 
-    std::string sql = PrepareSQL("UPDATE installed SET enabled=%d WHERE addonID='%s'", disable ? 0 : 1, addonID.c_str());
+    std::string sql =
+        PrepareSQL("UPDATE installed SET enabled=0, disabledReason=%d WHERE addonID='%s'",
+                   static_cast<int>(disabledReason), addonID.c_str());
+    m_pDS->exec(sql);
+    return true;
+  }
+  catch (...)
+  {
+    CLog::Log(LOGERROR, "%s failed on addon '%s'", __FUNCTION__, addonID.c_str());
+  }
+  return false;
+}
+
+bool CAddonDatabase::EnableAddon(const std::string& addonID)
+{
+  try
+  {
+    if (!m_pDB)
+      return false;
+    if (!m_pDS)
+      return false;
+
+    std::string sql =
+        PrepareSQL("UPDATE installed SET enabled=1, disabledReason=0 WHERE addonID='%s'", addonID.c_str());
```

Comment:
> The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

Reply:
Done

Reply:
no

Reply:
Apologies. Was just about to push and internet connection disappeared. Will push once it’s back.

Reply:
Now it's done ;)